node/src/components/block_synchronizer/execution_results_acquisition/tests.rs

```diff
@@ -0,0 +1,446 @@
+use assert_matches::assert_matches;
+use casper_types::{bytesrepr::ToBytes, testing::TestRng};
+use rand::Rng;
+
+use super::*;
+use crate::{
+    components::block_synchronizer::tests::test_utils::chunks_with_proof_from_data, types::Block,
+};
+
+const NUM_TEST_EXECUTION_RESULTS: u64 = 100000;
+
+#[test]
+fn execution_results_chunks_apply_correctly() {
+    let mut rng = TestRng::new();
+    let block = Block::random(&mut rng);
+
+    // Create chunkable execution results
+    let exec_results: Vec<ExecutionResult> = (0..NUM_TEST_EXECUTION_RESULTS)
+        .into_iter()
+        .map(|_| rng.gen())
+        .collect();
+    let test_chunks = chunks_with_proof_from_data(&exec_results.to_bytes().unwrap());
+    assert!(test_chunks.len() >= 3);
+
+    // Start off with only one chunk applied
+    let mut chunks: HashMap<u64, ChunkWithProof> = HashMap::new();
+    let first_chunk = test_chunks.first_key_value().unwrap();
+    let last_chunk = test_chunks.last_key_value().unwrap();
+    chunks.insert(*first_chunk.0, first_chunk.1.clone());
+
+    // Insert all the other chunks except the last; skip the first one since it should have been
+    // added already
+    for (index, chunk) in test_chunks.iter().take(test_chunks.len() - 1).skip(1) {
+        let apply_result = apply_chunk(
+            *block.hash(),
+            ExecutionResultsChecksum::Uncheckable,
+            chunks,
+            chunk.clone(),
+            None,
+        );
+
+        // Check the index of the next chunk that should be applied
+        chunks = assert_matches!(apply_result, Ok(ApplyChunkOutcome::NeedNext{chunks, chunk_count, next}) => {
+            assert_eq!(next, index + 1);
+            assert_eq!(chunk_count as usize, test_chunks.len());
+            chunks
+        });
+    }
+
+    // Apply the last chunk, and expect to get back the execution results
+    let apply_result = apply_chunk(
+        *block.hash(),
+        ExecutionResultsChecksum::Uncheckable,
+        chunks,
+        last_chunk.1.clone(),
+        None,
+    );
+    assert_matches!(apply_result, Ok(ApplyChunkOutcome::Complete{execution_results}) => {
+        assert_eq!(execution_results, exec_results);
+    });
+}
+
+#[test]
+fn single_chunk_execution_results_dont_apply_other_chunks() {
+    let mut rng = TestRng::new();
+    let test_chunks = chunks_with_proof_from_data(&[0; ChunkWithProof::CHUNK_SIZE_BYTES - 1]);
+    assert_eq!(test_chunks.len(), 1);
+
+    // We can't apply a chunk if the execution results are not chunked (only 1 chunk exists)
+    // Expect an error in this case.
+    let first_chunk = test_chunks.first_key_value().unwrap();
+
+    let apply_result = apply_chunk(
+        *Block::random(&mut rng).hash(),
+        ExecutionResultsChecksum::Uncheckable,
+        test_chunks.clone().into_iter().collect(),
+        first_chunk.1.clone(),
+        None,
+    );
+
+    assert_matches!(apply_result, Err(Error::InvalidChunkCount { .. }));
```

Comment:
I think we're only missing tests for `BlockHashMismatch` and `ChunkCountMismatch` error variants. Please consider extending the tests coverage.

Reply:
Added in https://github.com/casper-network/casper-node/pull/3698/commits/cf25d79e8b45387b0b5396422f2fa0d58627f3c4